File: lib/java/src/main/java/io/github/msgpact/internal/types/VHeaders.java
package io.github.msgpact.internal.types;

import java.util.Map;

public class VHeaders {
    public final String name;
    public final Map<String, VFieldDeclaration> requestHeaders;
    public final Map<String, VFieldDeclaration> responseHeaders;

    public VHeaders(
            final String name,
            final Map<String, VFieldDeclaration> requestHeaders,
            final Map<String, VFieldDeclaration> responseHeaders) {
        this.name = name;
        this.requestHeaders = requestHeaders;
        this.responseHeaders = responseHeaders;
    }
}

File: lib/ts/src/internal/types/VHeaders.ts
import { VFieldDeclaration } from './VFieldDeclaration';

export class VHeaders {
    name: string;
    requestHeaders: { [key: string]: VFieldDeclaration };
    responseHeaders: { [key: string]: VFieldDeclaration };

    constructor(
        name: string,
        requestHeaders: { [key: string]: VFieldDeclaration },
        responseHeaders: { [key: string]: VFieldDeclaration },
    ) {
        this.name = name;
        this.requestHeaders = requestHeaders;
        this.responseHeaders = responseHeaders;
    }
}
